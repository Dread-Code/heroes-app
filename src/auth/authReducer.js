import { LOGIN, LOGOUT } from "../types/types";

export const authReducer = (state = {}, action) => {
  switch (action?.type) {
    case LOGIN:
      return {
        ...action.payload,
        logged: true,
      };
    case LOGOUT:
      return {
        logged: false,
      };

    default:
      return state;
  }
};
