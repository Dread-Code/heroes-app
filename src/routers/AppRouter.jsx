import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { LoginScreen } from "../components/login/LoginScreen";
import { Marvelscreen } from "../components/marvel/Marvelscreen";
import { Navbar } from "../components/ui/NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/login" component={LoginScreen} />
          <Route exact path="/" component={Marvelscreen} />
        </Switch>
      </div>
    </Router>
  );
};
