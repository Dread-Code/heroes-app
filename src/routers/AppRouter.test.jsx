import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AppRouter } from "./AppRouter";
import { AuthContext } from "../auth/AuthContext";

describe("AppRouter", () => {
  const contextValue = { user: { logged: false }, dispatch: jest.fn() };

  test("should show the login if is not auth", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <AppRouter />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("should show the marvel component if is auth", () => {
    const contextValue = {
      user: { logged: true, name: "Lucas" },
      dispatch: jest.fn(),
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <AppRouter />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper.find(".navbar").exists()).toBe(true);
  });
});
