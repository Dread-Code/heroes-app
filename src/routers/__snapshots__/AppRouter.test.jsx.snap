// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter should show the login if is not auth 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      login
    </button>
  </div>
</div>
`;
